lib/zlib.js

```diff
@@ -21,14 +21,15 @@
 
 'use strict';
 
+const errors = require('internal/errors');
 const Buffer = require('buffer').Buffer;
 const Transform = require('_stream_transform');
 const { _extend } = require('util');
 const binding = process.binding('zlib');
 const assert = require('assert').ok;
-const kMaxLength = require('buffer').kMaxLength;
-const kRangeErrorMessage = 'Cannot create final Buffer. It would be larger ' +
-                           `than 0x${kMaxLength.toString(16)} bytes`;
+const {
+  kMaxLength
+} = require('buffer');
```

Comment:
Nit - `buffer` is actually required before as well and that could be combined by writing

```js
const { kMaxLength, Buffer } = require('buffer');
```

Reply:
I'll get to that later but feel free to push a commit if you'd like